lightning/src/ln/channel.rs

```diff
@@ -1946,9 +1946,24 @@ impl<Signer: Sign> Channel<Signer> {
 	/// an HTLC more than once or fulfill once and then attempt to fail after reconnect. We cannot,
 	/// however, fail more than once as we wait for an upstream failure to be irrevocably committed
 	/// before we fail backwards.
+	///
+	/// If we do fail twice, we debug_assert!(false) and return Ok(()). Thus, will always return
```

Comment:
Should be "Thus, we'll" or "Thus, this will"? Also `debug_assert!(false)` is a bit confusing at this point?
```suggestion
	/// If we do fail twice, we `debug_assert!(is_none)` and return `Ok(())`. Thus, we'll always return
```